scripts/o3de/ExportScripts/export_source_ios_xcode.py

```diff
@@ -0,0 +1,122 @@
+    :param ios_build_folder:                        The base output path of the generated Xcode Project file for iOS
+    :param should_build_tools:                      Option to build the export process dependent tools (AssetProcessor, AssetBundlerBatch, and dependencies)
+    :param should_build_all_assets:                 Option to process all the assets for the game
+    :param logger:                                  Optional logger to use to log the process and errors
+    """
+    
+    # Use a provided logger or get the current system one
+    if not logger:
+        logger = logging.getLogger()
+        logger.setLevel(logging.ERROR)
+
+    tools_build_folder_str = str(tools_build_folder)
+    ios_build_folder_str = str(ios_build_folder)
+
+    # Optionally build the toolchain needed to process the assets
+    if should_build_tools:
+        process_command(["cmake", "-B", tools_build_folder_str, '-G', "Xcode","-DLY_UNITY_BUILD=ON"], cwd=ctx.project_path)
+
+        process_command(["cmake", "--build", tools_build_folder_str, "--target", "Editor", "AssetProcessorBatch", "--config", "profile"],
+                    cwd=ctx.project_path)
+        
+    # Optionally process the assets
+    if not skip_asset_processing:
+        asset_processor_batch_path = exp.get_asset_processor_batch_path(tools_build_folder, True)
+        process_command([ str(asset_processor_batch_path), '--platforms=ios' ,
+                        '--project-path', ctx.project_path ], cwd=ctx.project_path)
+
+    # Generate the Xcode project file for the O3DE project
+    cmake_toolchain_path = ctx.engine_path / 'cmake/Platform/iOS/Toolchain_ios.cmake'
+
+    process_command(['cmake', '-B', ios_build_folder_str, '-G', "Xcode", f'-DCMAKE_TOOLCHAIN_FILE={str(cmake_toolchain_path)}', '-DLY_UNITY_BUILD=ON', '-DLY_MONOLITHIC_GAME=1'],
+                    cwd= ctx.project_path)
+
+
+    logger.info(f"Xcode project file should be generated now. Please check {ios_build_folder_str}")
+
+
+
+# This code is only run by the 'export-project' O3DE CLI command
+if "o3de_context" in globals():
+    global o3de_context
+    global o3de_logger
+
+    def parse_args(o3de_context: exp.O3DEScriptExportContext):
+        parser = argparse.ArgumentParser(
+                    prog=f'o3de.py export-project -es {__file__}',
+                    description="Exports a project as a generated iOS Xcode project in the project directory. "
+                                "In order to use this script, the engine and project must be setup and registered beforehand. ",
+                    epilog=exp.CUSTOM_CMAKE_ARG_HELP_EPILOGUE,
+                    formatter_class=argparse.RawTextHelpFormatter,
+                    add_help=False
+        )
+
+        default_tools_path = o3de_context.project_path / 'build/Mac'
+        default_ios_path = o3de_context.project_path / 'build/iOS'
```

Comment:
I think these build paths should match the default paths in the CMake presets which are:
```suggestion
        default_tools_path = o3de_context.project_path / 'build/mac_xcode'
        default_ios_path = o3de_context.project_path / 'build/ios_mono'
```
From: https://github.com/o3de/o3de/blob/development/cmake/Platform/Mac/CMakePresets.json#L76
And:
https://github.com/o3de/o3de/blob/development/cmake/Platform/iOS/CMakePresets.json#L28